pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -964,6 +964,9 @@ func (iqr *IQR) AsResult(qType structs.QueryType) (*structs.PipeSearchResponseOu
 	for i, record := range cValRecords {
 		recordsAsAny[i] = make(map[string]interface{})
 		for key, value := range record {
+			if !includeNulls && value.CVal == nil {
```

Comment:
nit: 
```suggestion
			if !includeNulls && value.IsNull() {
```
This handles a few more edge cases